clang-tools-extra/docs/clang-tidy/checks/readability/avoid-return-with-void-value.rst

```diff
@@ -0,0 +1,30 @@
+.. title:: clang-tidy - readability-avoid-return-with-void-value
+
+readability-avoid-return-with-void-value
+========================================
+
+Complains about statements returning expressions of type ``void``. It can be
+confusing if a function returns an expression even though its return type is
+``void``.
```

Comment:
consider: "Void function is intended to perform a task without producing a return value. Including a return value in such functions can lead to confusion and may miscommunicate the function's intended behavior." or something similar.